src/Microsoft.AspNetCore.Http.Features/IHttpResponseTrailersFeature.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    public interface IHttpResponseTrailersFeature
+    {
+        /// <summary>
+        /// The response trailers to send.
+        /// </summary>
+        IHeaderDictionary Trailers { get; set; }
+
+        /// <summary>
+        /// Indicates if the server has started writing the response trailers.
+        /// If true, the <see cref="Trailers"/> are now immutable, and <see cref="OnStarting"/> should no longer be called.
+        /// </summary>
+        bool HasStarted { get; }
+
+        /// <summary>
+        /// Registers a callback to be invoked just before writing the response trailers.
+        /// This is the last chance to modify the <see cref="Trailers"/>.
+        /// </summary>
+        /// <param name="callback">The callback to invoke before writing the trailers.</param>
+        /// <param name="state">The state to pass into the callback.</param>
+        void OnStarting(Func<object, Task> callback, object state);
```

Comment:
Nit: Maybe call this OnWritingResponseTrailers.

I get that my suggested name is a little redundant with the name of the interface, but I think the current naming is too similar to IHttpResponseFeature.OnStarting and could cause confusion.

Reply:
There's already an `.OnSendingHeaders`, maybe an `.OnSendingTrailerHeaders` would work?

Reply:
There's an `.OnSendingHeaders`? I think that's the OWIN key. It's called `OnStarting` in the ASP.NET Core abstractions, that's why I chose this name.

https://github.com/aspnet/HttpAbstractions/blob/b85ed9d5cd6cc174491bc84708093f16ddfc4e00/src/Microsoft.AspNetCore.Http.Features/IHttpResponseFeature.cs#L42-L49

Reply:
Ah. Yes I see what you mean. Shame it wasn't called something more like the owin feature then. Consistency would probably win here.